include/range/v3/view/linear_distribute.hpp

```diff
@@ -0,0 +1,183 @@
+              : view_facade<linear_distribute_integral_view<T>, finite>
+            {
+                static_assert(std::is_integral<T>(), "");
+            private:
+                friend range_access;
+                T from_, to_, spacing_;
+                constexpr T read() const
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const
+                {
+                    return from_ == to_ + spacing_;
+                }
+                constexpr bool equal(linear_distribute_integral_view const &that) const
+                {
+                    return that.from_ == from_ && that.to_ == to_ && that.spacing_ == spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void next()
+                {
+                    from_ += spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void prev()
+                {
+                    from_ -= spacing_;
+                }
+                RANGES_CXX14_CONSTEXPR void advance(std::ptrdiff_t n)
+                {
+                    RANGES_EXPECT(std::numeric_limits<T>::max() / spacing_ >= n);
+                    RANGES_EXPECT(to_ - from_ >= n * spacing_);
+                    from_ += n * spacing_;
+                }
+                constexpr std::ptrdiff_t distance_to(linear_distribute_integral_view const &that) const
+                {
+                    return (that.from_ - from_) / spacing_;
+                }
+            public:
+                linear_distribute_integral_view() = default;
+                linear_distribute_integral_view(T from, T to, std::ptrdiff_t n)
+                  : from_(from), to_(to) 
+                {
+                    RANGES_EXPECT(n > 0);
+                    RANGES_EXPECT(to_ >= from_);
+                    if (to_ == from_)
+                    {
+                        spacing_ = 0;   
+                    }
+                    else
+                    {
+                        RANGES_EXPECT(n > 1);
+                        spacing_ = (to_ - from_)/ static_cast<T>(n - 1);   
+                    }
+                }
+                constexpr std::size_t size() const noexcept
+                {
+                    return (to_ - from_) / spacing_;
+                }
+            };
+
+            template <typename T>
+            struct linear_distribute_floating_point_view
+              : view_facade<linear_distribute_floating_point_view<T>, finite>
+            {
+                static_assert(std::is_floating_point<T>(), "");
+            private:
+                friend range_access;
+
+                T from_, to_;
+                std::ptrdiff_t n_;
+                
+                constexpr T read() const
+                {
+                    return from_;
+                }
+                constexpr bool equal(default_sentinel) const
+                {
+                    return n_ == 0;
+                }
```

Comment:
> You need to add an `equal` that compares...

...in order to model `Sentinel<I, I>()` and `EqualityComparable<I>()`. Is there any motivation for doing so in addition to modeling `InputIterator<I>()` and `Sentinel<default_sentinel, I>()`?